Add e2e tests for events command

ardaguclu opened the pull request:
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
This PR adds new e2e tests for `kubectl events` command into 
e2e test suite.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

ardaguclu committed:
Add e2e tests for events command

ardaguclu commented:
/triage accepted

ardaguclu commented:
@soltysh would you mind looking at this new e2e test for `kubectl events` command?. Thanks.

ardaguclu commented on test/e2e/kubectl/kubectl.go:
```diff
@@ -1939,6 +1939,43 @@ metadata:
+		var podName string
+
+		ginkgo.BeforeEach(func() {
+			podName = "e2e-test-httpd-pod"
```
Maybe we should change generic pod name to something specific to events to prevent possible conflicts with others tests?

aojea replied:
Each e2e create its own namespace to avoid this, you can also leverage the garbage collector to delete the test resources, the framework delete the namespace after test has finished

aojea commented on test/e2e/kubectl/kubectl.go:
```diff
@@ -1939,6 +1939,43 @@ metadata:
+			framework.RunKubectlOrDie(ns, "delete", "pods", podName)
+		})
+
+		framework.ConformanceIt("should show event when pod is created ", func() {
```
You can't go directly to conformance, there are some requirements of stability, first starts as a normal e2e

ardaguclu replied:
Thanks a lot @aojea for review. Do you have any handy tests as an example(to understand what normal e2e looks like)?, I could not find `framework.e2e` in codebase.

ardaguclu replied:
I think, I understand what you mean, just running with `ginkgo.It`

aojea commented on test/e2e/kubectl/kubectl.go:
```diff
@@ -1939,6 +1939,43 @@ metadata:
+			}
+
+			ginkgo.By("show started event for this pod")
+			events := framework.RunKubectlOrDie(ns, "alpha", "events", "--for=pod/"+podName)
```
Conformance are only for GA

ardaguclu committed:
Run events tests as normal e2e instead conformance

Conformance tests are only for GA features. Since `kubectl events`
currently is in alpha stage, e2e tests for this command should be
run as standard e2e.

ardaguclu commented:
This PR has nothing to do with integration tests;

/test pull-kubernetes-integration

soltysh reviewed (approved):
/lgtm
/approve

soltysh commented:
/label tide/merge-method-squash
/priority important-longterm
/triage accepted
